doc/tutorials/space_shooter/app/lib/step6/main.dart

```diff
@@ -0,0 +1,231 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/events.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/game.dart';
+import 'package:flame/input.dart';
+import 'package:flame/parallax.dart';
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(GameWidget(game: SpaceShooterGame()));
+}
+
+class SpaceShooterGame extends FlameGame
+    with PanDetector, HasCollisionDetection {
+  late Player player;
+
+  @override
+  Future<void> onLoad() async {
+    final parallax = await loadParallaxComponent(
+      [
+        ParallaxImageData('stars_0.png'),
+        ParallaxImageData('stars_1.png'),
+        ParallaxImageData('stars_2.png'),
+      ],
+      baseVelocity: Vector2(0, -5),
+      repeat: ImageRepeat.repeat,
+      velocityMultiplierDelta: Vector2(0, 5),
+    );
+    add(parallax);
+
+    player = Player();
+    add(player);
+
+    add(
+      SpawnComponent(
+        factory: (index) {
+          return Enemy();
+        },
+        period: 1,
+        area: Rectangle.fromLTWH(0, 0, size.x, -Enemy.enemySize),
+      ),
+    );
+  }
+
+  @override
+  void onPanUpdate(DragUpdateInfo info) {
+    player.move(info.delta.global);
+  }
+
+  @override
+  void onPanStart(DragStartInfo info) {
+    player.startShooting();
+  }
+
+  @override
+  void onPanEnd(DragEndInfo info) {
+    player.stopShooting();
+  }
+}
+
+class Player extends SpriteAnimationComponent
+    with HasGameReference<SpaceShooterGame> {
+  late final TimerComponent _bulletSpawner;
+
+  @override
+  Future<void> onLoad() async {
+    await super.onLoad();
+
+    animation = await game.loadSpriteAnimation(
+      'player.png',
+      SpriteAnimationData.sequenced(
+        amount: 4,
+        stepTime: .2,
+        textureSize: Vector2(32, 48),
+      ),
+    );
+
+    position = game.size / 2;
+    width = 100;
+    height = 150;
+    anchor = Anchor.center;
+
+    _bulletSpawner = TimerComponent(
```

Comment:
SpawnComponent!

Reply:
This needs to be changed in all steps, can we merge this as is since the bullet was introduced in a previous step. And once this is merged, I will set a follow up to change on all steps.